### ISSUE TITLE
How to split `_RawTextIterableDataset`

### ISSUE THREAD

**Issue by KickItLikeShika** [pytorch/text#1265]
## ❓ Questions and Help
I am trying to move from using `legacy` and use new provided features, i was doing this:
```
from torchtext import legacy
TEXT = legacy.data.Field(lower=True, batch_first=True)
LABEL = legacy.data.LabelField(dtype=torch.float)
train_data, test_data = legacy.datasets.IMDB.splits(TEXT, LABEL, root='/tmp/imdb/')
train_data, valid_data = train_data.split(split_ratio=0.8, random_state=random.seed(SEED))
```
But now i want to split train_data, how can i do that?
```
from torchtext.datasets import IMDB
train_iter, test_iter = IMDB(split=('train', 'test'))
# I need to split train_iter into train_iter and valid_iter
```

**Comment by zhangguanheng66** [comment]
It's an iterator so I don't think you can split/shuffle it. I think, it's worth an option to set up the offset or the beginning of line. So for the valid set, you can start from a different line. cc @cpuhrsch @parmeet

**Comment by zhangguanheng66** [comment]
As a temporary solution, you can list all the items from the iterator and split them (if the dataset can fit the memory).
```
train_data = list(train_iter)
```

**Comment by KickItLikeShika** [comment]
Thanks! but i think if we did that we can not get it back to `_RawTextIterableDataset`, right?

**Comment by parmeet** [comment]
@KickItLikeShika Thank you for bringing this up. We will discuss this request internally and get back to you shortly!

**Comment by parmeet** [comment]
We have migrated our datasets on top of [torchdata datapipes](https://github.com/pytorch/data). @ejguan  I wonder if there is an elegant way to split datapipe into two random disjoint datapipes?

**Comment by enigdata** [comment]
could you provide code example?  split the train_iter of iMDB into train and validation
